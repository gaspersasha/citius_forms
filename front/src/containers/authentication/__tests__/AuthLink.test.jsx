import React from 'react';
import { AuthLink } from '..';

const props = {
  openModal: jest.fn(),
  type: 'login',
};

describe('<AuthLink />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<AuthLink {...props}>Link</AuthLink>);
  });

  it('should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call function', () => {
    wrapper.find('span').simulate('click');
    expect(props.openModal).toHaveBeenCalled();
  });
});
