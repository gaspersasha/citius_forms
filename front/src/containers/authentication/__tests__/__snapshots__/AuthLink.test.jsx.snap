// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthLink /> should match snapshots 1`] = `
<AuthenticationLink
  className=""
  dataId=""
  openModal={[MockFunction]}
  type="login"
>
  <li
    className="leaf"
  >
    <span
      className="link"
      data-id=""
      onClick={[MockFunction]}
      role="button"
      tabIndex={0}
    >
      Link
    </span>
  </li>
</AuthenticationLink>
`;
